doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -195,19 +285,548 @@ Mail chapel-developers@ for review and merge
+
+Move a file/dir:
+
+.. code-block:: bash
+
+    git mv orig/path/a.txt new/path/to/b.txt
+
+    # similar to:
+    svn move orig/path/a.txt new/path/to/b.txt
+
+Copy a file/dir and stage target for commit:
+
+.. code-block:: bash
+
+    cp <src> <target>
+    git add <target>
+
+    # similar to:
+    svn copy <src> <target>
+
+Get the status of files/dirs (staged and unstaged):
+
+.. code-block:: bash
+
+    git status
+
+    # similar to:
+    svn status
+
+Get the diff of unstaged changes:
+
+.. code-block:: bash
+
+    git diff
+
+    # similar to:
+    svn diff
+
+Get the diff of staged changes (those that were staged with ``git add``):
+
+.. code-block:: bash
+
+    git diff --cached
+
+Backing out unstaged changes:
+
+.. code-block:: bash
+
+    git checkout path/to/file/a.txt
+
+    # similar to:
+    svn revert path/to/file/a.txt
+
+Committing staged changes:
+
+.. code-block:: bash
+
+    git commit [-m <message>]
+
+    # similar to:
+    svn commit [-m <message>]
+
+Bring in changes from the main Chapel project:
+
+.. code-block:: bash
+
+    git fetch upstream
+    git merge upstream/master
+
+    # or:
+    git pull upstream <branch_name>
+
+    # with feature branch checked out:
+    git merge [--no-ff] upstream/master
+
+If there are conflicts, you will be asked to resolve them. Once the affected
+files have been fixed, stage them with ``git add``, and then call ``git
+commit`` to finish the merge process.
+
+If you want to understand the changes that occurred upstream, see
+`Read commit messages`_ below.
+
+Fixing a commit message:
+
+.. code-block:: bash
+
+    git commit --amend
+
+Note: This should only ever be done to commits that **have not been pushed** to
+a remote repository.
+
+Un-do the last commit (leaving changed files in your working directory)
+
+.. code-block:: bash
+
+    git reset --soft HEAD~1
+
+Note: This should only ever be done to commits that **have not been pushed** to
+a remote repository.
+
+.. _Read commit messages:
+
+Read commit messages for changes from the main Chapel project
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To view only the commits that happened on master (in other words, the old svn
+commits and the merge commits for pull requests):
+
+.. code-block:: bash
+
+    git log --first-parent
+
+    # or with line wrapping
+    git log --first-parent | less
+
+    # or including files changed
+    git log --first-parent -m --stat
+
+    # or similar to svn log
+    git log --first-parent -m --name-status
+
+More logging commands are described in `Other logging commands`_ below.
+
+.. _How to push:
+
+How to push
++++++++++++
+
+.. code-block:: bash
+
+    git push origin <branch_name>
+
+    # or if you don't like typing your complicated branch name,
+    # you can use this command to push the current branch:
+    git push origin HEAD
+
+    # if you forgot your branch name, you can get it by running
+    git branch
+
+    # it is the starred one...
```

Comment:
This could say that `push -f` is necessary if you've modified history on your feature branch (but see modifying history section)